third-party/linshare/src/main/java/org/apache/james/linshare/LinshareConfiguration.java

```diff
@@ -0,0 +1,104 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.linshare;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Objects;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+public class LinshareConfiguration {
+
+    public static class Builder {
+
+        @FunctionalInterface
+        public interface RequireUrl {
+            RequireUsername url(URL url);
+
+            default RequireUsername urlAsString(String url) throws MalformedURLException {
+                return url(new URL(url));
+            }
+        }
+
+        public interface RequireUsername {
```

Comment:
We should require a long lived token.

https://ci.linagora.com/linagora/lgs/linshare/products/linshare-github/blob/master/documentation/EN/configuration/how-to-use-jwt.md